hera_pspec/pspecdata.py

```diff
@@ -585,54 +596,105 @@ def C_model(self, key, model='empirical', time_index=None):
             subsequent indices specify the baseline index, in _key2inds format.
 
         model : string, optional
-            Type of covariance model to calculate, if not cached. options=['empirical', 'dsets']
+            Type of covariance model to calculate, if not cached. Options=['empirical', 'dsets', 'autos',...]
             How the covariances of the input data should be estimated.
             in 'dsets' mode, error bars are estimated from user-provided
             per baseline and per channel standard deivations. If 'empirical' is
             set, then error bars are estimated from the data by calculating the
             channel-channel covariance of each baseline over time and
             then applying the appropriate linear transformations to these
-            frequency-domain covariances.
+            frequency-domain covariances. If 'autos' is set, the covariances of the input data
+            over a baseline is estimated from the autocorrelations of the two antennas over channel bandwidth 
+            and integration time. 
 
         time_index : integer, compute covariance at specific time-step in dset
-                       only supported if mode == 'dsets'
+                       supported if mode == 'dsets' or 'autos'
 
+        known_cov : dicts of covariance matrices
+            Covariance matrices that are not calculated internally.
+        
         Returns
         -------
-        C : array-like
+        C : array-like, (Ntimes x spw_Nfreqs x spw_Nfreqs)
             Covariance model for the specified key.
         """
         # type check
         assert isinstance(key, tuple), "key must be fed as a tuple"
         assert isinstance(model, (str, np.str)), "model must be a string"
-        assert model in ['empirical', 'dsets'], "didn't recognize model {}".format(model)
+
         # parse key
         dset, bl = self.parse_blkey(key)
         key = (dset,) + (bl,)
         if model == 'dsets':
-            assert isinstance(time_index, int), "time_index must be integer if cov-model=dsets"
+            assert isinstance(time_index, int), "time_index must be integer if cov-model==dsets"
+            # add model to key
             Ckey = key + (model, time_index)
 
         elif model == 'empirical':
-            assert time_index is None, "time_indexing not supported in empirical covariances"
             # add model to key
             Ckey = key + (model,)
 
+        elif model == 'autos':
+            assert isinstance(time_index, int), "time_index must be integer if cov-model==autos"
+            # add model to key
+            Ckey = key + (model, time_index)
+
+        else:
+            if known_cov == None:
+                raise ValueError("didn't recognize model {}".format(model))
+            else:
+                assert isinstance(time_index, int), "time_index must be integer if cov-model=autos"
```

Comment:
what is this catching? `model` (typo) is not autos b/c it passed over that conditional. and why is the Ckey so different here? See my general comment about standardizing Ckey